### PR TITLE
Nexus writer

### PR DESCRIPTION
**Pull request by Modularius**
A functional tool which creates a NeXus file upon the start of a run, and appends event data to it when frame-aggregated or digitizer event list messages are created.

### CODE REVIEW

File: nexus-writer/diagrams/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+!*.svg
```
Comment (DanNixon) [inline nexus-writer/diagrams/.gitignore:1]: Is there a need to have SVG in the root `.gitignore`?

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: nexus-writer/diagrams/.gitignore (hunk 1 above)
Comment (Modularius) [reply nexus-writer/diagrams/.gitignore]: No, that folder is a duplicate of docs in fact

Comment (DanNixon) [reply nexus-writer/diagrams/.gitignore]: You will now need to alter the root `.gitignore`.

**Review by DanNixon — CHANGES_REQUESTED** [review]
[see above]

**Pull request merged**